apis/python/src/tiledbsoma/collection.py

```diff
@@ -183,7 +184,8 @@ def __delitem__(self, key: str) -> None:
     def __iter__(self) -> Iterator[str]:
         if self._cached_values is None:
             self._load_tdb_group_cache()
-        assert self._cached_values is not None
+        if self._cached_values is None:
```

Comment:
`assert`s are fine for internal sanity checks, so in this situation (`assert self._cached_values is not None`) it is fine.

Reply:
@thetorpedodog if this _ever_ goes off for a user I want a clear message "You didn't do anything wrong -- we did" surfaced to them. Hence the "internal error" wording.